WooCommerce/Classes/ViewRelated/Orders/Order Details/Shipping Labels/Create Shipping Label Form/Customs/ShippingLabelCustomsFormInputViewModel.swift

```diff
@@ -65,12 +75,83 @@ final class ShippingLabelCustomsFormInputViewModel: ObservableObject {
         self.restrictionComments = customsForm.restrictionComments
         self.itn = customsForm.itn
         self.items = customsForm.items
+        self.destinationCountry = destinationCountry
         self.allCountries = countries
         self.currency = currency
         self.itemViewModels = customsForm.items.map { .init(item: $0, countries: countries, currency: currency) }
+
+        resetContentExplanationIfNeeded()
+        resetRestrictionCommentsIfNeeded()
+    }
+}
+
+// MARK: - Validation
+//
+extension ShippingLabelCustomsFormInputViewModel {
+    var missingContentExplanation: Bool {
+        if contentsType == .other, contentExplanation.trimmingCharacters(in: .whitespacesAndNewlines).isEmpty {
+            return true
+        }
+        return false
+    }
+
+    var missingRestrictionComments: Bool {
+        if restrictionType == .other, restrictionComments.trimmingCharacters(in: .whitespacesAndNewlines).isEmpty {
+            return true
+        }
+        return false
+    }
+
+    var missingITNForDestination: Bool {
+        if itnValidationRequired && itn.isEmpty {
+            return true
+        }
+        return false
+    }
+
+    var missingITNForClassesAbove2500usd: Bool {
+        // TODO: check for accumulated value of each tariff number.
+        return false
+    }
+
+    var invalidITN: Bool {
+        if itn.isNotEmpty,
+           itn.range(of: Constants.itnRegex, options: .regularExpression, range: nil, locale: nil) == nil {
+            return true
+        }
+        return false
     }
 }
 
-// MARK: - Helper methods
+// MARK: - Private helpers
 //
-private extension ShippingLabelCustomsFormInputViewModel {}
+private extension ShippingLabelCustomsFormInputViewModel {
+
+    /// Reset content explanation if content type is not Other.
+    ///
+    func resetContentExplanationIfNeeded() {
+        $contentsType
+            .filter { $0 != .other }
+            .map { _ -> String in "" }
+            .assign(to: &$contentExplanation)
+    }
+
+    /// Reset restriction comments if restriction type is not Other.
+    ///
+    func resetRestrictionCommentsIfNeeded() {
+        $restrictionType
+            .filter { $0 != .other }
+            .map { _ -> String in "" }
+            .assign(to: &$restrictionComments)
+    }
+}
+
+private extension ShippingLabelCustomsFormInputViewModel {
+    enum Constants {
+        static let itnRegex = "^(?:(?:AES X\\d{14})|(?:NOEEI 30\\.\\d{1,2}(?:\\([a-z]\\)(?:\\(\\d\\))?)?))$"
```

Comment:
nit: Do you think it's worth adding a comment with details about where this regex comes from in the plugin, for future reference?